src/api/context.js

```diff
@@ -0,0 +1,64 @@
+import * as Cookie from '../cookie';
+import * as Http from '../http';
+import Navigator from '../navigator';
+
+// consts
+import STATUS from '../status_codes';
+
+const DEFAULT_HOST = 'https://api.radar.io';
+
+class Context {
+  static getContext(callback) {
+    Navigator.getCurrentPosition((status, { latitude, longitude }) => {
+      if (status !== STATUS.SUCCESS) {
+        callback(status);
+      }
+
+      this.getContextForLocation({ latitude, longitude },
+        (status, context) => {
+          callback(status, context);
+          return;
+        }
+      )
+    });
+  }
+
+  static getContextForLocation({ latitude, longitude }, callback) {
+    const publishableKey = Cookie.getCookie(Cookie.PUBLISHABLE_KEY);
+
+    if (!publishableKey) {
+      callback(STATUS.ERROR_PUBLISHABLE_KEY);
+
+      return;
+    }
+
+    const queryParams = {
+      coordinates: `${latitude},${longitude}`,
+    };
+
+    const host = Cookie.getCookie(Cookie.HOST) || DEFAULT_HOST;
+    const url = `${host}/v1/context`;
+    const method = 'GET';
+    const headers = {
+      Authorization: publishableKey,
+    };
+
+    const onSuccess = (response) => {
+      try {
+        response = JSON.parse(response);
+
+        callback(STATUS.SUCCESS, response.context);
+      } catch (e) {
+        callback(STATUS.ERROR_SERVER);
+      }
+    };
+
+    const onError = (error) => {
+      callback(error);
+    };
```

Comment:
same with the `onSuccess` / `onError` handlers - we can probably pass one callback to `Http.request` and it'll send up the response code accordingly

Reply:
Yeah, even getting the host. I feel like the call to request should just be:
```js
Http.request('GET', '/v1/context', queryParams, callback)
```